docs/sources/get-started/labels/bp-labels.md

```diff
@@ -20,6 +20,9 @@ Too many label value combinations leads to too many streams. The penalties for t
 
 To avoid those issues, don't add a label for something until you know you need it! Use filter expressions (`|= "text"`, `|~ "regex"`, …) and brute force those logs. It works -- and it's fast.
 
+If you often extract a label from a log line at query time, the label has a high cardinality, and extracting that label is expensive in terms of performance; consider extracting the label on the client side and
+attach it as metadata to log lines using a [non-indexed label]({{< relref "./non-indexed-labels" >}}).
```

Comment:
Is "extract" the verb we want here? It means to remove.  I did some revising, I think this captures what you're trying to say here, but feel free to edit if I'm misunderstanding.
```suggestion
Cardinality is the overall count of values for one label. If you often search for a label that applies to many log lines, that label has a high cardinality. Cardinality can have a major impact on querying where, the higher the cardinality, the more items are required to be iterated over during a query. This makes high cardinality labels expensive in terms of performance; consider extracting the label on the client side and attaching it as metadata to log lines using a [non-indexed label]({{< relref "./non-indexed-labels" >}}).
```

Reply:
> Cardinality is the overall count of values for one label. If you often search for a label that applies to many log lines, that label has a high cardinality. Cardinality can have a major impact on querying where, the higher the cardinality, the more items are required to be iterated over during a query

I think this isn't quite right in this context. We use stream labels and the time span to reduce the number of logs we need to process. Now, this is talking about labels parsed at query time. So if a log matches a parser, the value for the label parsed is added to the result labels. So we'll still need to process all the logs. 

I think the original paragraph is more correct than the one you suggested. I agree _extract_ may not be the best term here. I'd instead use _parse_ here. Wdyt?

```suggestion
If you often parse a label from a log line at query time, the label has a high cardinality, and extracting that label is expensive in terms of performance; consider extracting the label on the client side
attaching it as metadata to log lines using a [non-indexed label]({{< relref "./non-indexed-labels" >}}).
```

Reply:
I'll go with your rewording (not sure what the etiquette is here for committing suggestions, but I'm assuming that's your role as the PR owner).